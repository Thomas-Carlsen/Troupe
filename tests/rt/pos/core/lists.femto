let fun map f list = 
    case list of 
       [] => []
      |  (x::xs) => (f x) :: (map f xs)

    fun foldl f y [] = y
      | foldl f y (x::xs) = foldl f (f (x,y)) xs

    fun add (x, y) = x + y


    fun reverse ls =
      let fun rev [] a = a
            | rev (x::xs) a = rev xs (x::a)

      in rev ls []
      end

    fun range n =
     let fun rng 0 = []
          |  rng k = k :: (rng (k-1))
      in reverse (rng n)
     end


    fun fib x =
          if x > 2
          then fib (x - 1) + fib (x - 2)
          else 1


in
   foldl ( fn (x,y) => x + y) 0 
   ( map fib (range 10))
end
