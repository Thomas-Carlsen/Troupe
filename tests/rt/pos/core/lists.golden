2019-03-07T12:08:15.783Z [RTM] [32minfo[39m: Skipping network creation. Observe that all external IO operations will yield a runtime error.
>>> Main thread finished with value: 143@{}%{}
